bench/wtperf/wtperf.c

```diff
@@ -1362,6 +1362,62 @@ checkpoint_worker(void *arg)
     return (WT_THREAD_RET_VALUE);
 }
 
+static WT_THREAD_RET
+flush_tier_worker(void *arg)
+{
+    CONFIG_OPTS *opts;
+    WTPERF *wtperf;
+    WTPERF_THREAD *thread;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    uint32_t i;
+    int ret;
+
+    thread = (WTPERF_THREAD *)arg;
+    wtperf = thread->wtperf;
+    opts = wtperf->opts;
+    conn = wtperf->conn;
+    session = NULL;
+
+    if ((ret = conn->open_session(conn, NULL, opts->sess_config, &session)) != 0) {
+        lprintf(wtperf, ret, 0, "open_session failed in flush_tier thread.");
+        goto err;
+    }
+
+    while (!wtperf->stop) {
+        /* Break the sleep up, so we notice interrupts faster. */
+        for (i = 0; i < opts->tiered_flush_interval; i++) {
+            sleep(1);
+            if (wtperf->stop)
+                break;
+        }
+        /* If the workers are done, don't bother with a final call. */
+        if (wtperf->stop)
+            break;
+
+        wtperf->flush = true;
+        if ((ret = session->flush_tier(session, NULL)) != 0) {
```

Comment:
Maybe include `sync=on`?  By analogy, `checkpoint_worker()` waits for the checkpoint to finish then waits for the `checkpoint_interval` to start the next checkpoint.  So maybe it sesms more natural for `flush_tier` to work the same way.

Reply:
`sync=on` is the default, do you think it needs to be explicit? Also, this new function was originally a copy of the `checkpoint_worker` function, and it is waiting the flush interval to start the next flush_tier. So it does work the exact same way as checkpoint. I must be misunderstanding your comment in some way. Can you be specific on what change you want to see regarding the interval?